starter-code/scripts/articleView.js

```diff
@@ -16,50 +16,82 @@ articleView.populateFilters = function() {
   });
 };
 
+
 articleView.handleAuthorFilter = function() {
   $('#author-filter').on('change', function() {
-  if ($(this).val()) {
-      /* TODO: If the slect box changes to an option that has a value, we should:
-          1. Hide all of the articles
-          2. Fade in only the articles that match based on on the author
-            that was aselected. Hint: use an attribute selector to find
-            those articles that match the value, and then fade them in.
-        */
-  } else {
-    /* Otherwise, we should:
-        1. Show all the articles except the template */
-  }
+    if ($(this).val()) {
+      $('article').hide();
+      $(`article[data-author="${ $(this).val() }"]`).fadeIn(1000)
+      // $('article[data-author=" ' + $(this).val() + '"]').show();
+      //     /* TODO: DONE: If the slect box changes to an option that has a value, we should:
+      //         1. Hide all of the articles
+      //         2. Fade in only the articles that match based on on the author
+      //           that was aselected. Hint: use an attribute selector to find
+      //           those articles that match the value, and then fade them in.
+      //       */
+    } else {
+      $('article').not('.template').show();
+    }
     $('#category-filter').val('');
   });
 };
 
 articleView.handleCategoryFilter = function() {
-  /* TODO: Just like we do for #author-filter above, we should also handle
-  change events on the #category-filter element. Be sure to reset the
-  #author-filter while you're at it! */
+  $('#category-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').hide();
+      $(`article[data-category="${ $(this).val() }"]`).fadeIn(1000)
+
+      /* TODO: Done: Just like we do for #author-filter above, we should also handle
+      change events on the #category-filter element. Be sure to reset the
+      #author-filter while you're at it! */
+    } else {
+      $('article').show();
+    }
+    $('#category-filter').val('');
+    $('#author-filter').val('');
+
+  });
 };
 
 articleView.handleMainNav = function () {
   $('.main-nav').on('click', '.tab', function() {
-    /* TODO:
-      1. Hide all of the .tab-content sections
-      2. Fade in the single .tab-content section that is
-        associated with the .tab element's data-content attribute.
-    */
-  });
+    console.log($(this).data('content'));
+  if ($(this).data('content')) {
+    $('.tab-content').hide();
+     $('#' + $(this).data('content')).fadeIn(1000);
+  };
+})
+  // /* TODO: DONE:
+  //   1. Hide all of the .tab-content sections
+  //   2. Fade in the single .tab-content section that is
+  //     associated with the .tab element's data-content attribute.
+  // */
+  // });
   $('.main-nav .tab:first').click();
 };
 
 articleView.setTeasers = function() {
   $('.article-body *:nth-of-type(n+2)').hide();
+  $('.read-on').on('click', function(eve) {
+    eve.preventDefault()
+    $(this).parent().find('.read-on').text('Show Less')
```

Comment:
This is a good start towards the stretch goal, but it isn't going to toggle between 'Read On' and 'Show Less'. :)